### PR TITLE
fix: maximum image dimensions on preview generation and avatar upload

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/Core/Controller/AvatarControllerTest.php [hunk 1]
```diff
@@ -293,6 +293,25 @@ public function testPostAvatarFile() {
...
+		$copyRes = \copy(\OC::$SERVERROOT.'/tests/data/pixel.jpg', $fileName);
+		$this->assertTrue($copyRes);
+
+		//Create request return
+		$reqRet = ['error' => [0], 'tmp_name' => [$fileName], 'size' => [\filesize(\OC::$SERVERROOT.'/tests/data/testimage.jpg')]];
+		$this->request->expects($this->once())->method('getUploadedFile')->willReturn($reqRet);
+
+		$response = $this->avatarController->postAvatar(null);
+
+		//On correct upload always respond with the notsquare message
```
Comment (phil-davis) [inline tests/Core/Controller/AvatarControllerTest.php:308]: comment needs adjusting

**Review by phil-davis — COMMENTED** [review]
Changelog needed.

File: core/Controller/AvatarController.php [hunk 2]
```diff
@@ -215,6 +215,15 @@ public function postAvatar($path) {
 			$image = new \OC_Image();
 			$image->load($handle);
 			$image->fixOrientation();
+			// don't generate previews on images with too big dimensions
```
Comment (jvillafanez) [inline core/Controller/AvatarController.php:218]: Comment needs adjustment: this is avatar upload, not preview generation

**Comment by jvillafanez** [comment]
I guess we don't really want a migration in order to homogenize the avatars because it will be expensive. Any alternative or workaround?

**Comment by DeepDiver1975** [comment]
> I guess we don't really want a migration in order to homogenize the avatars because it will be expensive.

No - the issue is only during generating as far as I can say.

**Review by phil-davis — APPROVED** [review]

**Review by jvillafanez — APPROVED** [review]

**Pull request merged**